docker/Dockerfile-dev

```diff
@@ -1,7 +1,5 @@
-FROM stactools-aster:latest
+FROM stacutils/stactools:0.2.1a1-dev
 
-RUN conda install -c conda-forge pandoc
-
-COPY . /src/stactools-aster
-RUN pip install -r /src/stactools-aster/requirements-dev.txt
-WORKDIR /src/stactools-aster
+WORKDIR /opt/aster
+COPY . ./
+RUN pip install --no-deps -e .
```

Comment:
I noticed in https://github.com/stac-utils/stactools/pull/136 that the different images were built with `--target` rather than keeping separate {Dockerfile|Dockerfile-dev} files? Should/can we use the same structure in downstream repos as well (aka just have one Dockerfile)?

Reply:
The downstream ones Dockerfiles don't actually share anything between them, so we could go either way.
The upstream ones share intermediate images so it's easier to keep them in one Dockerfile